Common/Product/SharedProject/ProcessOutput.cs

```diff
@@ -184,23 +184,20 @@ sealed class ProcessOutput : IDisposable {
                     redirector,
                     quoteArgs,
                     outputEncoding,
-                    errorEncoding
-                );
-            }
-
-            var psi = new ProcessStartInfo(filename);
-            if (quoteArgs) {
-                psi.Arguments = string.Join(" ",
-                    arguments.Where(a => a != null).Select(QuoteSingleArgument));
-            } else {
-                psi.Arguments = string.Join(" ", arguments.Where(a => a != null));
-            }
-            psi.WorkingDirectory = workingDirectory;
-            psi.CreateNoWindow = !visible;
-            psi.UseShellExecute = false;
-            psi.RedirectStandardError = !visible || (redirector != null);
-            psi.RedirectStandardOutput = !visible || (redirector != null);
-            psi.RedirectStandardInput = !visible;
+                    errorEncoding);
+            }
+
+            var psi = new ProcessStartInfo("cmd.exe") { 
+                Arguments = string.Format(@"/S /C pushd {0} & {1} {2}",
```

Comment:
does pushd have a timeout? What happens if there is no network access or the drive is inaccessible?